dapps/onboarding.vue/src/components/twin-sign-up/twin-sign-up.ts

```diff
@@ -0,0 +1,433 @@
+/*
+  Copyright (C) 2018-present evan GmbH.
+
+  This program is free software: you can redistribute it and/or modify it
+  under the terms of the GNU Affero General Public License, version 3,
+  as published by the Free Software Foundation.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+  See the GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program. If not, see http://www.gnu.org/licenses/ or
+  write to the Free Software Foundation, Inc., 51 Franklin Street,
+  Fifth Floor, Boston, MA, 02110-1301 USA, or download the license from
+  the following URL: https://evan.network/license/
+*/
+
+// vue imports
+import Vue from 'vue';
+import Component, { mixins } from 'vue-class-component';
+import { Prop } from 'vue-property-decorator';
+import axios from 'axios';
+
+// evan.network imports
+import { EvanForm, EvanFormControl, getDomainName, } from '@evan.network/ui-vue-core';
+import * as bcc from '@evan.network/api-blockchain-core';
+import * as dappBrowser from '@evan.network/ui-dapp-browser';
+import SignUp from '../sign-up/sign-up.ts';
+
+interface TwinDBCPForm extends EvanForm {
+  name: EvanFormControl;
+  description: EvanFormControl;
+}
+
+@Component({ })
+export default class TwinSignUp extends mixins(SignUp) {
+  /**
+   * Form for getting twin name / description.
+   */
+  twinDbcpForm: TwinDBCPForm = null;
+
+  /**
+   * Formular for getting twin metadata entry.
+   */
+  metadataForm: EvanForm = null;
+
+  /**
+   * Formular for gettin list entry data.
+   */
+  listForm: EvanForm = null;
+
+  /**
+   * Setup twin forms.
+   */
+  created() {
+    // makes a field required
+    const requiredValidator = (vueInstance: TwinSignUp, form: EvanForm, control: EvanFormControl) => {
+      return control.value && control.value.length !== 0;
+    }
+
+    // base ui specs, that will be passed to the dynamic controls
+    const uiSpecs = { attr: { label: '' } };
+    const control = { value: '', uiSpecs, };
+    const metadataControl = JSON.parse(JSON.stringify(control));
```

Comment:
Why not just write it like this?
```suggestion
    const metadataControl = { value: '', uiSpecs, };
```